packages/xo-web/src/common/xo/utils.js

```diff
@@ -24,8 +24,12 @@ export const getDefaultNetworkForVif = (vif, destHost, pifs, networks) => {
   return destNetworkId
 }
 
-export const getDefaultMigrationNetwork = (destHost, pools, pifs) => {
+export const getDefaultMigrationNetwork = (intraPool, destHost, pools, pifs) => {
   const migrationNetwork = pools[destHost.$pool].otherConfig['xo:migrationNetwork']
+  if (intraPool) {
+    return migrationNetwork
+  }
```

Comment:
I think you still want to iterate through the PIFs in this case, since we still want to make sure that the network exists, that there's a PIF attached to it, etc.